llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -9352,6 +9353,27 @@ SDValue TargetLowering::expandAVG(SDNode *N, SelectionDAG &DAG) const {
     }
   }
 
+  if (Opc == ISD::AVGFLOORU && SVT == MVT::i128) {
```

Comment:
Don't limit this to i128 - better off using a !isTypeLegal(VT) check

Reply:
understood, thanks!